enhancements/update/upgrades-blocking-on-ack.md

```diff
@@ -0,0 +1,162 @@
+This enhancement adds a generic upgrade blocking mechanism for blocking minor level upgrades.
+This generic upgrade blocker can consist of one or more items each of which will block a minor level upgrade.
+Each generic upgrade blocker item will be associated with some well-defined task to be completed by an administrator.
+Once the task has been completed the administrator will acknowledge completion and its associated item will no longer block minor level upgrades.
+
+## Motivation
+
+OCP 4.9 uses kubernetes v1.22 which is removing a large number of [APIs][k8s-1.22-removed-apis].
+We suspect that many customer clusters are still using these APIs through helm charts, operators, generic workloads, administrative tools, etc.
+We believe there is no effective way for a Cluster Operator to detect the use of these APIs by external applications.
+Relying solely on automated API detection to block upgrades could result in blocking for APIs that are not actively used any longer or not blocking because an application was not active during the check.
+Having a human in the loop to drive the analysis and make the final decision is the surest way to prevent customers from upgrading from 4.8.z to 4.9.z and encountering missing API related breakage.
+
+### Goals
+
+* Acknowledgment requires the explicit update of a cluster resource by an administrator.
+* For the initial release in 4.8.z, administrator acknowledgment of API removal will be required before a minor level upgrade to 4.9 is allowed.
+* Upgrade blocking mechanism must be generic such that it can be repurposed in the future.
+* It must be clear what an administrator is acknowledging.
+* Administrators can bypass the upgrade blocking mechanism by using the existing CVO "force" implementation.
+* The UI will clearly indicate when this upgrade blocking mechanism is blocking upgrades, why, and what needs to be done to clear the block.
+* This upgrade blocking mechanism must be compatible with SD, ACM, IKS/ROKS, Hive, and Hypershift.
+
+### Non-Goals
+
+## Proposal
+
+### General Implementation
+
+CVO's existing framework for blocking minor level upgrades will be leveraged.
+The existing framework periodically checks upgradeable conditions.
+A new [upgradeable check method][upgradeableCheck-interface] will be added to the existing framework to implement this upgrade blocking mechanism.
+Use of this existing framework also allows this new upgrade blocking mechanism to be bypassed by using the `force` upgrade option.
+
+The new method's logic will be driven by predefined configmap key names each of which represents a potential minor level upgrade blocker.
+The key name will include an OCP major/minor verion number that indicates to which OCP upgrade target version the key applies.
+For example, a key name containing `4.9` would only apply to an OCP cluster currently at version 4.8 and can therefore only block 4.8 to 4.9 upgrades.
```

Comment:
This may be an implementation detail that we need not call out here, but is the controller strictly parsing the defined key names for the version or is it just expected to be defined in the hardcoded gating code? If we're relying on the key name perhaps we need to define the format for the names?

Reply:
Defined key name format.